src/Dependency/TagDependency.php

```diff
@@ -108,7 +110,12 @@ private function getStoredTagTimestamps(CacheInterface $cache, array $tags): ite
      */
     private static function buildCacheKey(string $tag): string
     {
-        return md5(json_encode([__CLASS__, $tag]));
+        $jsonTag = json_encode([__CLASS__, $tag]);
+        if ($jsonTag === false) {
```

Comment:
Is it possible that `json_encode([__CLASS__, $tag])` will return `false`?

Reply:
It’s possible according to the documentation.

Reply:
`false` may be returned if value is impossible to express as JSON (like resource). I doubt that array of two strings is such case ;).

Reply:
I think there are more reasons of failure
```
                JSON_ERROR_NONE => 'No error',
                JSON_ERROR_DEPTH => 'Maximum stack depth exceeded',
                JSON_ERROR_STATE_MISMATCH => 'State mismatch (invalid or malformed JSON)',
                JSON_ERROR_CTRL_CHAR => 'Control character error, possibly incorrectly encoded',
                JSON_ERROR_SYNTAX => 'Syntax error',
                JSON_ERROR_UTF8 => 'Malformed UTF-8 characters, possibly incorrectly encoded'
```

Reply:
Only `JSON_ERROR_UTF8` or `JSON_ERROR_UTF16` seems to be possible. In that case using `json_last_error_msg()` as exception message may be more useful.